src/service.ts

```diff
@@ -0,0 +1,13 @@
+import { Client } from "./client";
+
+export class Service {
+    
+    client: Client;
```

Comment:
TypeScript has an (optional) shorned syntax for this pattern. It compiles down to the same thing, but less typing.

```suggestion
    constructor(public client: Client) {}
```

Reply:
Also, you may want to add `readonly` there too, so if I do:
```ts
const s = new Service(client);
s.client = new Client(); // ❌ wouldn't compile, since `client` is set as `readonly`
```